stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -713,8 +712,10 @@ extension Unicode.Scalar.Properties {
       }
     }
     // TODO: replace `len` with `count`
-    return array[..<len].withUnsafeBufferPointer {
-      return String._uncheckedFromUTF16($0)
+    return array.withUnsafeBufferPointer { (bufPtr: UnsafeBufferPointer<UInt64>) -> String in
+        return UnsafeBufferPointer(rebasing: bufPtr[..<len]).withMemoryRebound(to: UInt16.self) {
```

Comment:
From my reading, `len` is the `UInt16` count, but it's being used here with the `UInt64` storage. Do I have that backwards?

In addition, `withMemoryRebound` prohibits changing the size of the bound type, so this isn't a legal call. See https://developer.apple.com/documentation/swift/unsafebufferpointer/2950055-withmemoryrebound

Reply:
fixed